src/modules/musig/tests_impl.h

```diff
@@ -1078,52 +951,239 @@ void musig_test_vectors(void) {
+        0x03,
+        0x74, 0x96, 0xA3, 0xCC, 0x86, 0x92, 0x6D, 0x45,
+        0x2C, 0xAF, 0xCF, 0xD5, 0x5D, 0x25, 0x97, 0x2C,
+        0xA1, 0x67, 0x5D, 0x54, 0x93, 0x10, 0xDE, 0x29,
+        0x6B, 0xFF, 0x42, 0xF7, 0x2E, 0xEE, 0xA8, 0xC9,
+    };
+    const unsigned char sk[32] = {
+        0x7F, 0xB9, 0xE0, 0xE6, 0x87, 0xAD, 0xA1, 0xEE,
+        0xBF, 0x7E, 0xCF, 0xE2, 0xF2, 0x1E, 0x73, 0xEB,
+        0xDB, 0x51, 0xA7, 0xD4, 0x50, 0x94, 0x8D, 0xFE,
+        0x8D, 0x76, 0xD7, 0xF2, 0xD1, 0x00, 0x76, 0x71,
+    };
+    const unsigned char msg[32] = {
+        0xF9, 0x54, 0x66, 0xD0, 0x86, 0x77, 0x0E, 0x68,
+        0x99, 0x64, 0x66, 0x42, 0x19, 0x26, 0x6F, 0xE5,
+        0xED, 0x21, 0x5C, 0x92, 0xAE, 0x20, 0xBA, 0xB5,
+        0xC9, 0xD7, 0x9A, 0xDD, 0xDD, 0xF3, 0xC0, 0xCF,
+    };
+    const unsigned char *pk[2] = { vec_pk[0], vec_pk[1] };
+
+    {
+        const unsigned char sig_expected[32] = {
+            0x68, 0x53, 0x7C, 0xC5, 0x23, 0x4E, 0x50, 0x5B,
+            0xD1, 0x40, 0x61, 0xF8, 0xDA, 0x9E, 0x90, 0xC2,
+            0x20, 0xA1, 0x81, 0x85, 0x5F, 0xD8, 0xBD, 0xB7,
+            0xF1, 0x27, 0xBB, 0x12, 0x40, 0x3B, 0x4D, 0x3B,
+        };
+        musig_test_vectors_sign_helper(&keyagg_cache, &fin_nonce_parity, sig, state, agg_pubnonce, sk, msg, pk, 0);
+        /* TODO: remove when test vectors are not expected to change anymore */
+        /* int k, l; */
+        /* printf("const unsigned char sig_expected[32] = {\n"); */
+        /* for (k = 0; k < 4; k++) { */
+        /*     printf("    "); */
+        /*     for (l = 0; l < 8; l++) { */
+        /*         printf("0x%02X, ", sig[k*8+l]); */
+        /*     } */
+        /*     printf("\n"); */
+        /* } */
+        /* printf("};\n"); */
+
+        /* This is a test where the combined public key point has an _odd_ y
+         * coordinate, the signer _is not_ the second pubkey in the list and the
+         * nonce parity is 1. */
+        CHECK(musig_test_pk_parity(&keyagg_cache) == 1);
+        CHECK(!musig_test_is_second_pk(&keyagg_cache, sk));
+        CHECK(fin_nonce_parity == 1);
+        CHECK(memcmp(sig, sig_expected, 32) == 0);
+    }
+    {
+        const unsigned char sig_expected[32] = {
+            0x2D, 0xF6, 0x7B, 0xFF, 0xF1, 0x8E, 0x3D, 0xE7,
+            0x97, 0xE1, 0x3C, 0x64, 0x75, 0xC9, 0x63, 0x04,
+            0x81, 0x38, 0xDA, 0xEC, 0x5C, 0xB2, 0x0A, 0x35,
+            0x7C, 0xEC, 0xA7, 0xC8, 0x42, 0x42, 0x95, 0xEA,
+        };
+        musig_test_vectors_sign_helper(&keyagg_cache, &fin_nonce_parity, sig, state, agg_pubnonce, sk, msg, pk, 1);
+
+       /* This is a test where the aggregate public key point has an _even_ y
+        * coordinate, the signer _is_ the second pubkey in the list and the
+        * nonce parity is 0. */
+        CHECK(musig_test_pk_parity(&keyagg_cache) == 0);
+        CHECK(musig_test_is_second_pk(&keyagg_cache, sk));
+        CHECK(fin_nonce_parity == 0);
+        CHECK(memcmp(sig, sig_expected, 32) == 0);
     }
```

Comment:
Here's one more test case that puts the signer in the final position. Conveniently, it's also a case where the parity of the agg pubkey is different from the parity of the nonce!

```c
    {
        const unsigned char sig_expected[32] = {
            0x0D, 0x5B, 0x65, 0x1E, 0x6D, 0xE3, 0x4A, 0x29,
            0xA1, 0x2D, 0xE7, 0xA8, 0xB4, 0x18, 0x3B, 0x4A,
            0xE6, 0xA7, 0xF7, 0xFB, 0xE1, 0x5C, 0xDC, 0xAF,
            0xA4, 0xA3, 0xD1, 0xBC, 0xAA, 0xBC, 0x75, 0x17,
        };
        musig_test_vectors_sign_helper(&keyagg_cache, &fin_nonce_parity, sig, state, agg_pubnonce, sk, msg, pk, 2);

       /* This is a test where the aggregate public key point has an _odd_ y
        * coordinate, the signer _is not_ the second pubkey in the list and the
        * nonce parity is 0. */
        CHECK(musig_test_pk_parity(&keyagg_cache) == 1);
        CHECK(!musig_test_is_second_pk(&keyagg_cache, sk));
        CHECK(fin_nonce_parity == 0);
        CHECK(memcmp(sig, sig_expected, 32) == 0);
    }
```

Reply:
Added that one